Sources/OpenAPIAsyncHTTPClient/AsyncHTTPClientTransport.swift

```diff
@@ -62,26 +62,27 @@ public struct AsyncHTTPClientTransport: ClientTransport {
         /// The HTTP client used for performing HTTP calls.
         public var client: HTTPClient
 
-        /// The default shared HTTP client.
-        ///
-        /// This is a workaround for the lack of a shared client
-        /// in AsyncHTTPClient. Do not use this value directly, outside of
-        /// the `Configuration.init(client:timeout:)` initializer, as it will
-        /// likely be removed in the future.
-        private static let sharedClient: HTTPClient = .init()
-
         /// The default request timeout.
         public var timeout: TimeAmount
 
         /// Creates a new configuration with the specified client and timeout.
         /// - Parameters:
         ///   - client: The underlying client used to perform HTTP operations.
-        ///     Provide nil to use the shared internal client.
         ///   - timeout: The request timeout, defaults to 1 minute.
-        public init(client: HTTPClient? = nil, timeout: TimeAmount = .minutes(1)) {
-            self.client = client ?? Self.sharedClient
+        public init(client: HTTPClient = .shared, timeout: TimeAmount = .minutes(1)) {
+            self.client = client
             self.timeout = timeout
         }
+
+        /// Creates a new configuration with the specified client and timeout.
+        /// - Parameters:
+        ///   - client: The underlying client used to perform HTTP operations.
+        ///     Provide nil to use the shared client.
+        ///   - timeout: The request timeout, defaults to 1 minute.
+        @available(*, deprecated, renamed: "init(client:timeout:)") @_disfavoredOverload public init(
+            client: HTTPClient? = nil,
+            timeout: TimeAmount = .minutes(1)
+        ) { self.init(client: client ?? .shared, timeout: timeout) }
```

Comment:
Why deprecate this one vs. just changing `client ?? Self.sharedClient` to `client ?? .shared`?

Reply:
This one has client as an optional HTTPClient, but that doesn't make sense anymore now that we have a public symbol that we can use as the default value in the parameter.

Reply:
I don't disagree but I also don't know if it's worth doing as the observed impact for users is either zero (used the default or always provided their own non-optional `HTTPClient`) or negative (they explicitly passed `client: nil` or had their own logic for deciding whether to use the default or not so passed in `HTTPClient?`. In the latter case they'd need to move to a new API to get the same behaviour.

That said this is a very minor thing and I don't mind either way, I just thought it was worth noting.

Reply:
Yeah `client: nil` is what I'm trying to prevent people from doing now, and emitting a warning is the best we can do within this major version. All the other cases should not get a warning, they should pick up the new initializer automatically, I believe.

Thanks for checking me on this, though 🙂

Reply:
OOI, does the `renamed:` play well with `@_disfavoredOverload`. Does it make sense? Gonna be a very confusing thing here since the symbol names are the same? Maybe just an informative message is better?

Reply:
Yeah good point, I'll remove the `renamed` and add a descriptive message instead.

Reply:
Done.